genty/genty.py

```diff
@@ -129,7 +129,7 @@ def _expand_repeats(test_functions):
                     dataprovider,
                     repeat_suffix,
                 )
-        elif dataset:
+        elif dataset or dataprovider:
```

Comment:
I wondered that as well when stepping through the code. But just went with the flow. Changing it to "else" does work. All tests continue to pass.